File: Dynamic_Programming/01KnapSack/target_sum.java


class target_sum {

    public int findTargetSubsets(int [] nums, int S) {
        int totalSum = 0;
        for(int n : nums) {
            totalSum += n;
        }
        // if (s+totalSum) is odd, we cant find a subset with sum equal to (s + totalSum) /2
        if (totalSum < 0 || (S + totalSum) % 2 == 1) {
            return 0;
        }

        return
    }
}

File: EDUCATIVE/Dynamic_Programming/01KnapSack/Target_sum.py

def find_target_sum(nums, target):
    totalSum = sum(nums)

    if totalSum <= target or (target + totalSum)/2 == 1:
        return 0
    
    return target_sum_bottom_up(nums, totalSum, target)



def target_sum_bottom_up(nums, totalSum, target):
     n = len(nums)

     dp = [ [0 for i in range(target)] for y in range(n)]

     